pkg/apiserver/user/auth.go

```diff
@@ -54,34 +56,41 @@ type TokenResponse struct {
 	Expire time.Time `json:"expire"`
 }
 
-func (f *authenticateForm) Authenticate(tidbForwarder *tidb.Forwarder) (*utils.SessionUser, error) {
-	// TODO: Support non TiDB auth
-	if !f.IsTiDBAuth {
-		return nil, ErrSignInUnsupportedAuthType.New("unsupported auth type, only TiDB auth is supported")
-	}
-	db, err := tidbForwarder.OpenTiDB(f.Username, f.Password)
-	if err != nil {
-		if errorx.Cast(err) == nil {
+func (f *authenticateForm) Authenticate(tidbForwarder *tidb.Forwarder, etcdClient *clientv3.Client) (*utils.SessionUser, error) {
+	if !*f.IsTiDBAuth {
+		//currently only allow root username
+		if f.Username != "root" {
```

Comment:
why do we only allow root user even in KvAuth mode?

Reply:
we need to allow user to supply a custom super user name as described in the UCP body: "supply a super user login username and password"